### PR TITLE
Automatically convert SVG to PDF images in the latex builder

### PR DESCRIPTION
**Pull request by alyjak**
This adds the ability to automatically convert svg images to pdfs using Inkscape when using the latex builder. This gets around the fact that latexpdf cannot natively handle svg images.

This change should be non invasive -- no new dependencies are required. Only if Inkscape is already installed will the code spawn a subprocess to call it to convert svgs into pdfs. If Inkscape is not available, the existing code should be unaffected besides an extra console print that lets users know that if they install Inkscape they will get this new feature.

Notes:
- I haven't tested this with windows.
- Some users might find the extra print out to be spammy. Let me know what you think and I'd be happy to remove it.
- I've run all the unit tests with and without Inkscape installed -- they all seem to pass in python 3.5, but when run in python 2.7 a couple seem to fail but they look to be unrelated to this change
- The unit tests have different behavior based off of whether Inkscape is installed. If its not, test_environment.py `test_images()` works the same as it did. If it is installed, `test_images()` expects to see `svgimg.svg: svgimg_svg2pdf.pdf` in its image map.
- I didn't see any place in the docs that needed to be updated for this change. Let me know if you think there's any documentation that needs updating.
- I didn't think this was a big enough change to update `CHANGES`, I also wasn't sure which section should be updated anyways.

### CODE REVIEW

**Comment by tk0miya** [comment]
I'll take a look later.

refs: #2166, #2852

**Comment by anarcat** [comment]
"This branch has conflicts that must be resolved" - otherwise what's the status on this?

**Comment by tk0miya** [comment]
Sorry for late response.

I feel this is very nice PR, but it would be nice if other builders also support SVG format.
So I'd like to improve this PR to use this feature from other builders.
I will try to improve this until next major release. If I could not, I will merge this as it is.

Note: It would be great if we can support other tools for image conversion.

**Comment by anarcat** [comment]
the HTML builder already supports SVG. i suspect it *may* work with ePUB, depending on the reader, but I haven't tested that directly.

while it would be great to have more generic support for arbitrary file formats, this can quickly get into a very complicated project: witness imagemagick and how many security vulnerability it has. i would personally keep only to a few formats (PNG, JPG, SVG) and convert SVGs as necessary.

an alternative would be simply to refuse SVG files and explicitely require those to be converted by the user, since, in the end, that is what we effectively do for them...

also note that certain tools use [librsvg](https://en.wikipedia.org/wiki/Librsvg)'s [rsvg-convert](https://manpages.debian.org/rsvg-convert) to convert files. it may be better than Inkscape, because there are less dependencies...

in any case, conflicts need to be resolved here first. @alyjak are you still working on this PR?

thanks all!

**Comment by tk0miya** [comment]
>the HTML builder already supports SVG. i suspect it may work with ePUB, depending on the reader, but I haven't tested that directly.

We already have many builders. And some builders also does not support SVG images. For example devhelp, texinfo and other builders. Also 3rd party builders might not support them.
It is ineffecient to add SVG support to each builder. I prefer to add conversion mechanism to sphinx-core.

Anyway, we still have 4-6 months duration until next major release; Sphinx-1.6. So I would not merge this for a while even if resolved.

**Comment by alyjak** [comment]
Sorry for the late response, i'll submit the updated PR later tonight

**Comment by jfbu** [comment]
memo: for mac os x, a note could be added that users should install a compiled binary, because the bundled (from http://www.inkscape.org) Mac OS X application working under XQuartz requires full path filenames when executed from the command line for input and output https://bugs.launchpad.net/inkscape/+bug/1449251

edit: perhaps this changes with [0.92 release](https://inkscape.org/en/download/mac-os/) and its [macports install](https://inkscape.org/en/download/mac-os/#alternative_install_0.92) (not tested)

File: sphinx/builders/latex.py [hunk 1]
```diff
@@ -56,6 +57,22 @@ def init(self):
...
+                retcode = subprocess.call(['inkscape', '--version'],
+                                          stdout=FNULL,
+                                          stderr=subprocess.STDOUT)
+                if retcode == 0:
+                    self.supported_image_types.append('image/svg+xml')
+                    self.svg_support = True
+            except OSError:
+                pass
+        if not self.svg_support:
+            self.info("Latex builder initialized without svg support. You can"
```
Comment (mitya57) [inline sphinx/builders/latex.py]: I would print this line only if the project actually contains SVG images (i.e. move it below).

Also, please capitalize “LaTeX” and “SVG” properly.

**Comment by alyjak** [comment]
@mitya57, thanks for the feedback, I have updated the warning to capitalize LaTeX and SVG correctly, and I have moved the warning to only display if inkscape is not present and there are SVG images in the build

**Comment by mitya57** [comment]
Thank you. In our project we would like to use some SVG images too, so I fully support the idea of this PR.

**Comment by tk0miya** [comment]
I just proposed a new extension; `sphinx.ext.imgconverter` in #3653.
I used Imagemagick to do that. But it also allows you to convert images with your tools if you make an extension.
Please give a comment if possible.
Thanks,

**Comment by tk0miya** [comment]
Now I merged #3653 instead.
Thank you for proposal. Actually, this affected to my PR.

**Comment by mitya57** [comment]
@tk0miya Sorry, but imagemagick is not a replacement for inkscape. Imagemagick [rasterizes](http://www.imagemagick.org/discourse-server/viewtopic.php?t=6458#post_content19406) the SVG image, so the text layers are lost, and the resulting image is not scalable.

The correct way to use SVG images with LaTeX is converting them to PDF with Inkscape or other program that understands vector graphics.

**Comment by tk0miya** [comment]
I know, inkscape is a better choice. But there are many image converter tools. To use them, #3653 provides a way to enhance image converter. So you are able to make a converter using inkscape.
I think `sphinx.ext.imgconverter` is a reference implementation of image converters.

**Comment by mitya57** [comment]
OK, that is better. But it would be still nice if SVG to PDF using inkscape was available out of the box.

**Comment by tk0miya** [comment]
Absolutely, but we don't have enough time to maintain all of extensions as you know. So our basic strategy is make sphinx extensible and slim.

**Comment by Balletie** [comment]
Found this thread, made this to convert my SVG images to PDF using `rsvg-convert`:

https://gist.github.com/Balletie/a3ccf434f8ec492e47fff4589225d5cb

Just put it somewhere in `sys.path` so that it can be imported by Sphinx.

*Edit*: added inkscape one too

**Pull request closed without merging**